eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Models/ResourceMetadata.cs

```diff
@@ -11,5 +12,7 @@ internal record ResourceMetadata(
         ResourceScope ResourceScope,
         IReadOnlyList<ResourceMethod> Methods,
         string? SingletonResourceName,
-        string? ParentResourceId);
+        string? ParentResourceId,
+        IReadOnlyDictionary<string, InputClient> MethodToClientMap,
```

Comment:
I am not sure if this is a best way that we put this dictionary in the metadata - looks like we could have a map in our input library or output library for this cache?